pyxem/signals/diffraction_vectors.py

```diff
@@ -60,66 +60,51 @@
 """
 
 
-def _find_max_length_peaks(peaks):
-    """Worker function for generate_marker_inputs_from_peaks.
+def reverse_pos(peaks, ind=2):
+    """Reverses the position of the peaks in the signal."""
+    new_data = peaks[:, ind - 2 : ind]
+    return new_data[:, ::-1]
 
-    Parameters
-    ----------
-    peaks : :class:`~pyxem.diffraction_vectors.DiffractionVectors`
-        Identified peaks in a diffraction signal.
-
-    Returns
-    -------
-    longest_length : int
-        The length of the longest peak list.
 
-    """
-    length_of_longest_peaks_list = 0
-    with peaks.unfolded(unfold_navigation=True, unfold_signal=False):
-        for array in peaks.data:
-            if array.shape[0] > length_of_longest_peaks_list:
-                length_of_longest_peaks_list = array.shape[0]
-    return length_of_longest_peaks_list
+class ColumnSlicer:
+    def __init__(self, signal):
+        self.signal = signal
 
+    def __getitem__(self, item):
+        if isinstance(item, str):
+            item = self.signal.column_names.index(item)
+        elif isinstance(item, int):
+            pass
+        else:
+            raise ValueError("item must be a string or an int")
 
-def generate_marker_inputs_from_peaks(peaks):
-    """Takes a peaks (defined in 2D) object from a STEM (more than 1 image) scan
-    and returns markers.
-
-    Parameters
-    ----------
-    peaks : :class:`~pyxem.diffraction_vectors.DiffractionVectors`
-        Identifies peaks in a diffraction signal.
+        item = [
+            item,
+        ]
+        slic = self.signal.map(
+            lambda x, item: x[..., item], item=item, inplace=False, ragged=True
```

Comment:
Since `item` is constant, why not set it in the lambda argument list? That would be a bit more familiar to Python programmers: `lambda x, item=item[0]: ...`

Reply:
@sivborg Actually I don't know if that works. The `map` function has a bunch of passing between functions and it specifically requires that `kwargs` are passed to the function.  At least I think that is the case.